packages/atomic/src/components/result-lists/atomic-folded-result-list/atomic-folded-result-list.tsx

```diff
@@ -0,0 +1,169 @@
+
+  /**
+   * TODO: KIT-452 Infinite scroll feature
+   * Whether to automatically retrieve an additional page of results and append it to the
+   * current results when the user scrolls down to the bottom of element
+   */
+  private enableInfiniteScroll = false;
+  private renderingFunction?: (result: FoldedResult) => HTMLElement = undefined;
+
+  /**
+   * A list of non-default fields to include in the query results, separated by commas.
+   * The default fields sent in a request are: 'date', 'author', 'source', 'language', 'filetype', 'parents', ‘urihash’, ‘objecttype’, ‘collection’, ‘permanentid’ 'ec_price', 'ec_name', 'ec_description', 'ec_brand', 'ec_category', 'ec_item_group_id', 'ec_shortdesc', 'ec_thumbnails', 'ec_images', 'ec_promo_price', 'ec_in_stock', 'ec_cogs', and 'ec_rating'.
+   */
+  @Prop({reflect: true}) public fieldsToInclude = '';
+  /**
+   * The spacing of various elements in the result list, including the gap between results, the gap between parts of a result, and the font sizes of different parts in a result.
+   */
+  @Prop({reflect: true}) density: ResultDisplayDensity = 'normal';
+  /**
+   * The expected size of the image displayed in the results.
+   */
+  @Prop({reflect: true}) imageSize?: ResultDisplayImageSize;
+  /**
+   * @deprecated use `imageSize` instead.
+   */
+  @Prop({reflect: true}) image: ResultDisplayImageSize = 'icon';
+  /**
+   * TODO:
+   */
+  @Prop({reflect: true}) public collectionField?: string;
+  /**
+   * TODO:
+   */
+  @Prop({reflect: true}) public parentField?: string;
+  /**
+   * TODO:
+   */
+  @Prop({reflect: true}) public childField?: string;
+
+  /**
+   * Sets a rendering function to bypass the standard HTML template mechanism for rendering results.
+   * You can use this function while working with web frameworks that don't use plain HTML syntax, e.g., React, Angular or Vue.
+   *
+   * Do not use this method if you integrate Atomic in a plain HTML deployment.
+   *
+   * @param render
+   */
+  @Method() public async setRenderFunction(
+    render: (result: FoldedResult) => HTMLElement
+  ) {
+    this.renderingFunction = render;
+  }
+
+  @Listen('scroll', {target: 'window'})
+  handleInfiniteScroll() {
+    handleInfiniteScroll(this.enableInfiniteScroll, this.host, this.resultList);
+  }
+
+  public async initialize() {
+    await initializeResultList.call(this, this.initFolding.bind(this));
+  }
+
+  private initFolding(_: SearchEngine<{}>, options: FoldedResultListProps) {
+    const opts = {...options};
+
+    if (!opts.options) {
+      opts.options = {};
+    }
+    opts.options.folding = {};
+    if (this.collectionField) {
+      opts.options.folding.collectionField = this.collectionField;
+    }
+    if (this.parentField) {
+      opts.options.folding.parentField = this.parentField;
+    }
+    if (this.childField) {
+      opts.options.folding.childField = this.childField;
+    }
+
+    return buildFoldedResultList(this.bindings.engine, opts);
```

Comment:
Instead of all the above logic for options, I would prefer a bit of a one-liner like this:
```ts
    return buildFoldedResultList(
      this.bindings.engine,
      {
        ...options,
        collectionField: this.collectionField ?? 'foldingcollection',
        parentField: this.parentField ?? 'foldingparent',
        childField: this.childField ?? 'foldingchild',
      }
    );
```